### PR TITLE
Toilet

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR
Adds toilet. Things that you can do with it:

- [x] You can seat on toilet (sick!)
- [x] You can raise and lower toilet seat
- [x] You can unanchor toilet and run around with it
- [x] You can deconstruct toilet using welding tool to get some metal
- [x] Toilets are reagent containers with decent amount of toxic water
- [x] Using crowbar you can open cistern lid and stash something
- [x] You can swirlie yourself

I used almost same stash mechanic from potted plant. Added new SecretStash component that used both in toilet cistern and potted plant.

P.S. Inspired by: https://www.youtube.com/watch?v=hHqAvzGyy30

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
Why is the toilet water toxic by default?

Also, add a complex digestion system, a robust plumbing network and... AND....

**Comment by ike709** [comment]
@SweptWasTaken Assistants don't flush.

**Comment by PJB3005** [comment]
> You can swirlie yourself

But can you swirlie other people?

**Comment by komunre** [comment]
I like this idea. Maybe it will be good for Roleplay.

**Comment by Macoron** [comment]
> Why is the toilet water toxic by default?

Pee poo water bad

> But can you swirlie other people?

Nope, until grab will be implemented. 

> Also, washing sinks when?

Maybe in next PR

**Comment by mirrorcult** [comment]
> > But can you swirlie other people?
> 
> Nope, until grab will be implemented.

could just use click+drag plus a do_after

**Comment by Macoron** [comment]
> > > But can you swirlie other people?
> > 
> > 
> > Nope, until grab will be implemented.
> 
> could just use click+drag plus a do_after

This will buckle someone

**Comment by mirrorcult** [comment]
> 
> 
> > > > But can you swirlie other people?
> > > 
> > > 
> > > Nope, until grab will be implemented.
> > 
> > 
> > could just use click+drag plus a do_after
> 
> This will buckle someone

sad but I'm glad i can experience the gritty realism of buckling someone to a toilet

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        {
+            base.Initialize();
+            _secretStash = Owner.EnsureComponent<SecretStashComponent>();
+        }
+
+        public void MapInit()
+        {
+            // roll is toilet seat will be up or down
+            var random = IoCManager.Resolve<IRobustRandom>();
+            IsSeatUp = random.NextDouble() > 0.5f;
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:52]: could just do random.Prob

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            return true;
+        }
+
+        public void Examine(FormattedMessage message, bool inDetailsRange)
+        {
+            if (inDetailsRange && LidOpen)
+            {
+                if (_secretStash.HasItemInside())
+                {
+                    message.AddMarkup(Loc.GetString("There is [color=darkgreen]someting[/color] inside cistern!"));
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:119]: `someting` in the aaaiiiir tonight
```suggestion
                    message.AddMarkup(Loc.GetString("There is [color=darkgreen]something[/color] inside cistern!"));
```

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                victim.PopupMessage(selfMessage);
+
+                return SuicideKind.Asphyxiation;
+            }
+            else
+            {
+                var othersMessage = Loc.GetString("{0:theName} bashes themselves with {1:theName}!", victim, Owner);
+                victim.PopupMessageOtherClients(othersMessage);
+
+                var selfMessage = Loc.GetString("You bashes themselves with {0:theName}!", Owner);
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:161]:
```suggestion
                var selfMessage = Loc.GetString("You bashes yourself with {0:theName}!", Owner);
```

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        public SuicideKind Suicide(IEntity victim, IChatManager chat)
+        {
+            // check that victim even have head
+            if (victim.TryGetComponent<IBody>(out var body) &&
+                body.GetPartsOfType(BodyPartType.Head).Count != 0)
+            {
+                var othersMessage = Loc.GetString("{0:theName} sticks their head into {1:theName} and flushes it!", victim, Owner);
+                victim.PopupMessageOtherClients(othersMessage);
+
+                var selfMessage = Loc.GetString("You stick your head into {0:theName} and flushes it!", Owner);
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:151]:
```suggestion
                var selfMessage = Loc.GetString("You stick your head into {0:theName} and flush it!", Owner);
```

**Review by PaulRitter — COMMENTED** [review]
cannot stash any prying tools, currently fine imo since we dont really have intents rn

really nice to see these wellwritten content prs, a nice way to show off how easy it really is to implement new features :D

**Review by PaulRitter — CHANGES_REQUESTED** [review]
used wrong reviewkind, see above review for actual review

**Comment by Macoron** [comment]
@PaulRitter all issues should be resolved

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 5]
```diff
@@ -158,7 +158,7 @@ public SuicideKind Suicide(IEntity victim, IChatManager chat)
                 var othersMessage = Loc.GetString("{0:theName} bashes themselves with {1:theName}!", victim, Owner);
                 victim.PopupMessageOtherClients(othersMessage);
 
-                var selfMessage = Loc.GetString("You bashes themselves with {0:theName}!", Owner);
+                var selfMessage = Loc.GetString("You bashes yourself with {0:theName}!", Owner);
```
Comment (mirrorcult) [inline Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:162]: You bash yourself!

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        public sealed override string Name => "Toilet";
+
+        private const float PryLidTime = 1f;
+
+        private bool _isPrying = false;
+
+        [ViewVariables] public bool LidOpen { get; private set; }
+        [ViewVariables] public bool IsSeatUp { get; private set; }
+
+        [ViewVariables] private SecretStashComponent _secretStash = default!;
```
[the remark this answers is not in the document]
Comment (PaulRitter) [reply Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:41]: imo its fine

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        {
+            if (Owner.TryGetComponent(out SpriteComponent? sprite))
+            {
+                var state = string.Format("{0}_toilet_{1}",
+                    LidOpen ? "open" : "closed",
+                    IsSeatUp ? "seat_up" : "seat_down");
+
+                sprite.LayerSetState(0, state);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (Macoron) [reply Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:141]: Are you seriously want to create custom appearance class for a toilet?

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (Macoron) [reply Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:41]: Agree. Don't want to overengineer it with null checks. 
90% of the component doesn't make any sense if you can't hide something inside cistern.

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+                    message.AddMarkup(Loc.GetString("There is [color=darkgreen]something[/color] inside cistern!"));
+                }
+            }
+        }
+
+        public void ToggleToiletSeat()
+        {
+            IsSeatUp = !IsSeatUp;
+            UpdateSprite();
+        }
```
[the remark this answers is not in the document]
Comment (Macoron) [reply Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs]: Prying sound will play anyway (by using crowbar). Added amazing toilet seat sound.

File: Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Content.Server/GameObjects/Components/Watercloset/ToiletComponent.cs:147]: https://discord.com/channels/310555209753690112/770682801607278632/790229194319462430

**Comment by Macoron** [comment]
Added toilet visualizer and awesome seat sound. Wow

**Review by metalgearsloth — APPROVED** [review]

**Comment by metalgearsloth** [comment]
Game is complete.

**Pull request merged**